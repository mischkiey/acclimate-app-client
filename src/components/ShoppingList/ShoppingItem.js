import React from 'react';

function ShoppingItem(props) {
    return (
        <>
            <div className='list-group'>
                <button
                    className='y-btn'
                    onClick={(e) => props.handleCheckShoppingItem(e, props.user_id, props.user_shopping_item_id, props.user_shopping_item_completed)}
                >
                    <i className="material-icons">check</i>
                </button>
                <input
                    className=
                        {
                            (props.user_shopping_item_completed)
                                ? 'completed item-input' 
                                : 'item-input'
                        }
                    type='text'
                    defaultValue={props.user_shopping_item}
                    onChange={(e) => {props.handleEditShoppingItem(e, props.user_id, props.user_shopping_item_id)}}
                />
                <button
                    className='r-btn'
                    onClick={() => props.handleDeleteShoppingItem(props.user_shopping_item_id)}
                >
                    <i className="material-icons">delete</i>
                </button>
            </div>
        </>
    );
};

export default ShoppingItem;