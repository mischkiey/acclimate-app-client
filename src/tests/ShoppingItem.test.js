import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import { BrowserRouter } from 'react-router-dom';
import ShoppingItem from '../components/ShoppingList/ShoppingItem';

describe(`ShoppingItem Component`, () => {
    const item = {
        user_shopping_item: 'Logs'
    };

    describe(`Smoke test`, () => {
        it(`Renders without crashing`, () => {
            const div = document.createElement('div');
            ReactDOM.render(
                <BrowserRouter>    
                    <ShoppingItem {...item} />
                </BrowserRouter>
                , div);
            ReactDOM.unmountComponentAtNode(div);
        });
    });
    
    describe(`Snapshot test`, () => {
        it(`Renders the UI as expected`, () => {
            const tree = renderer
                .create(
                    <BrowserRouter>
                        <ShoppingItem {...item} />
                    </BrowserRouter>
                )
                .toJSON()
            expect(tree).toMatchSnapshot()
        });
    });
})