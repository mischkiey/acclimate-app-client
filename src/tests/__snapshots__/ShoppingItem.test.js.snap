// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ShoppingItem Component Snapshot test Renders the UI as expected 1`] = `
<div
  className="list-item-group"
>
  <button
    className="y-btn"
    onClick={[Function]}
  >
    <i
      className="material-icons"
    >
      check
    </i>
  </button>
  <input
    className="list-item-input"
    defaultValue="Logs"
    onChange={[Function]}
    type="text"
  />
  <button
    className="r-btn"
    onClick={[Function]}
  >
    <i
      className="material-icons"
    >
      delete
    </i>
  </button>
</div>
`;
